src/app/rmmapi/messagelist.service.ts

```diff
@@ -82,8 +82,8 @@ export class MessageListService {
                 if (msgFlagChange.flaggedFlag === true || msgFlagChange.flaggedFlag === false) {
                     this.messagesById[msgFlagChange.id].flaggedFlag = msgFlagChange.flaggedFlag;
                 }
-                // update both remote + local counts
-                this.refreshFolderList().then(folders => this.folderMessageCountSubject.next(this.getFolderCountsFor(folders)));
+                // update local results ASAP, schedule API results for later
+                this.refreshFolderCounts().then(() => this.refreshFolderList());
```

Comment:
hm, isnt refreshFolderList the thing that pulls in the new data for counting? how does this work backwards? - just tested with sync off and yes.. the read count doesnt update when I mark some messages as "unread"

Reply:
I'd expect it work backwards since `refreshFolderList()` calls `refreshFolderCounts()` again once it's done updating `this.folders`.

Reply:
... nevermind I was using an out of date backend!